src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiTextIndex.java

```diff
@@ -0,0 +1,207 @@
+
+    final var definitionOptions = new TextIndexDefinitionDesc.TextIndexDescOptions(analyzer);
+    final var definition =
+        new TextIndexDefinitionDesc(cqlIdentifierToJsonKey(targetColumn), definitionOptions);
+
+    return new IndexDesc<>() {
+      @Override
+      public String name() {
+        return cqlIdentifierToJsonKey(indexName);
+      }
+
+      @Override
+      public String indexType() {
+        return indexType.apiName();
+      }
+
+      @Override
+      public TextIndexDefinitionDesc definition() {
+        return definition;
+      }
+    };
+  }
+
+  /**
+   * Factory to create a new {@link ApiTextIndex} using {@link RegularIndexDefinitionDesc} from the
+   * user request.
+   */
+  private static class UserDescFactory
+      extends IndexFactoryFromIndexDesc<ApiTextIndex, TextIndexDefinitionDesc> {
+    @Override
+    public ApiTextIndex create(
+        TableSchemaObject tableSchemaObject, String indexName, TextIndexDefinitionDesc indexDesc) {
+
+      Objects.requireNonNull(tableSchemaObject, "tableSchemaObject must not be null");
+      Objects.requireNonNull(indexDesc, "indexDesc must not be null");
+
+      // for now, we are relying on the validation of the request deserializer that these values are
+      // specified userNameToIdentifier will throw an exception if the values are not specified
+      var indexIdentifier = userNameToIdentifier(indexName, "indexName");
+      var targetIdentifier = userNameToIdentifier(indexDesc.column(), "targetColumn");
+
+      var apiColumnDef = checkIndexColumnExists(tableSchemaObject, targetIdentifier);
+
+      // we could check if there is an existing index but that is a race condition, we will need to
+      // catch it if it fails - the resolver needs to set up a custom error mapper
+
+      // Text indexes can only be on text columns
+      if (apiColumnDef.type().typeName() != ApiTypeName.TEXT) {
+        throw SchemaException.Code.UNSUPPORTED_TEXT_INDEX_FOR_DATA_TYPES.get(
+            errVars(
+                tableSchemaObject,
+                map -> {
+                  map.put(
+                      "allColumns",
+                      errFmtApiColumnDef(tableSchemaObject.apiTableDef().allColumns()));
+                  map.put("unsupportedColumns", errFmt(apiColumnDef));
+                }));
+      }
+
+      Map<String, String> indexOptions = new HashMap<>();
+
+      // The analyzer is optional, if not specified default settings will be used
+      // (named analyzer "standard" in CQL).
+      // But further, if it is a StringNode, needs to be as a "raw" String, not a JSON String;
+      // but if ObjectNode, it must be encoded as JSON object.
+
+      JsonNode analyzerDef = indexDesc.options() == null ? null : indexDesc.options().analyzer();
+      if (analyzerDef == null) {
+        analyzerDef =
+            JsonNodeFactory.instance.textNode(
+                TableDescDefaults.CreateTextIndexOptionsDefaults.DEFAULT_NAMED_ANALYZER);
```

Comment:
Do we need to check if `analyzer` is a valid string here? If user passes in "foo", etc. wouldn't that cause a Cassandra error?

Reply:
It does, but we don't really have a way to check which analyzers exist (that is, which String values are legal), so will need to let Cassandra validate it. Same is true for Object options to a degree; we can do some validation but nothing thorough. Ultimately C* (only) knows full rules.